go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -716,10 +717,14 @@ func (qre *QueryExecutor) execSelect() (*sqltypes.Result, error) {
 				q.SetErr(err)
 			}
 		} else {
-			qre.logStats.QuerySources |= tabletenv.QuerySourceConsolidator
-			startTime := time.Now()
-			q.Wait()
-			qre.tsv.stats.WaitTimings.Record("Consolidations", startTime)
+			waiterCap := qre.tsv.config.ConsolidatorQueryWaiterCap
+			if waiterCap == 0 || c.WaiterCountOfTotal() <= waiterCap {
+				qre.logStats.QuerySources |= tabletenv.QuerySourceConsolidator
+				startTime := time.Now()
+				q.Wait()
```

Comment:
The waiter count increases on calling the `q.Wait()` method.
If you run 10s or 100s of request, they can still pass the check `c.WaiterCountOfTotal() <= waiterCap` before any new waiter not getting consolidated.
A concurrent test will help test if the waiter count exceeds the maximum allowed waiter.

I think the better place to update the wait counter will be in `create` call